components/server/src/workspace/workspace-starter.ts

```diff
@@ -173,14 +174,20 @@ export class WorkspaceStarter {
             startRequest.setSpec(spec);
             startRequest.setServicePrefix(workspace.id);
 
+            // we add additional information to the user to help with cluster selection
+            const euser: ExtendedUser = {
+                ...user,
+                getsMoreResources: await this.userService.userGetsMoreResources(user),
+            }
+
             // tell the world we're starting this instance
             let resp: StartWorkspaceResponse.AsObject | undefined;
-            let exceptInstallation: string[] = [];
             let lastInstallation = "";
-            for (let i = 0; i < 5; i++) {
+            const clusters = await this.clientProvider.getStartClusterSets(euser, workspace, instance);
+            for await (let cluster of clusters) {
```

Comment:
Why not `for (const cluster of clusters)`

Reply:
I see, an (async) iterator. Long time not seen. :-D

Reply:
`clusters` is an async iterator, hence requiring the `await`